src/design-system/components/button/Button.tsx

```diff
@@ -0,0 +1,93 @@
+import { forwardRef, ReactNode } from 'react'
+import {
+  Button as MuiButton,
+  CircularProgress,
+  ButtonProps as MuiButtonProps
+} from '@mui/material'
+
+import { cn } from '~/utils/cn'
+
+import '~scss-components/button/Button.scss'
+
+const sizes = ['xs', 'sm', 'md', 'lg'] as const
+
+const colors = [
+  'primary',
+  'tonal',
+  'text-primary',
+  'text-secondary',
+  'tonal-success',
+  'tonal-error'
+] as const
+
+type BaseButtonProps = {
+  loading?: boolean
+  size?: (typeof sizes)[number]
+  color?: (typeof colors)[number]
+  startIcon?: ReactNode
+  endIcon?: ReactNode
+  to?: string
+}
+
+export type ButtonProps = BaseButtonProps &
+  Omit<MuiButtonProps, keyof BaseButtonProps>
+
+type Ref = MuiButtonProps['ref']
+
+const Button = forwardRef(
```

Comment:
Add generic types to `forwardRef` function. So you will not have to type `Ref` manually. Also props at 40-48 lines will be typed as well.